modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpClientDiscoverySpiSelfTest.java

```diff
@@ -2579,14 +2581,16 @@ private void pauseResumeOperation(boolean isPause, AtomicBoolean... locks) {
         }
 
         /** {@inheritDoc} */
-        @Override protected void writeToSocket(Socket sock, TcpDiscoveryAbstractMessage msg, byte[] msgBytes,
```

Comment:
@aleksey-kurinov , I think we dont need this message-drop emulation at all (`TestTcpDiscoverySpi.oid writeToSocket(Socket sock, byte[] msgBytes, long timeout)`) because `TcpDiscoverySpi` doesn't send the message. It uses raw bites instead. This is the idea of the ticket. But several tests can fail after this change. It might become a challenge to understand what and why the tests simulate and how to fix them.